### ISSUE TITLE
Solidity by example demos

### ISSUE THREAD

**Issue by g-r-a-n-t** [ethereum/fe#499]
Implement the following Solidity example contracts in Fe:

- [Voting](https://docs.soliditylang.org/en/latest/solidity-by-example.html#voting)
- [Simple Open Auction](https://docs.soliditylang.org/en/latest/solidity-by-example.html#simple-open-auction)
- [Blind Auction](https://docs.soliditylang.org/en/latest/solidity-by-example.html#id2)
[- Safe Remote Purchase](https://docs.soliditylang.org/en/latest/solidity-by-example.html#safe-remote-purchase)

**Comment by WilfredTA** [comment]
For these, we will need to first add support for bytes as well as non-base types in struct fields right?

**Comment by g-r-a-n-t** [comment]
In places where Solidity uses `bytes32`, we would just use `u256`. I didn't see any spots where Solidity's dynamically sized `bytes` type is used, but I could be missing them.

**Comment by WilfredTA** [comment]
Good point w.r.t `bytes32`; I don't see any dynamically sized bytes after taking a second look, either.

So the only blocker, then, is ability to add non-primitives to arrays and structs it seems.
